include/glow/Partition/Partitioner.h

```diff
@@ -0,0 +1,78 @@
+/**
+ * Copyright (c) 2017-present, Facebook, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#ifndef GLOW_PARTITIONER_H
+#define GLOW_PARTITIONER_H
+
+#include "glow/Graph/Context.h"
+#include "glow/Graph/Graph.h"
+
+#include <map>
+#include <set>
+#include <string>
+
+namespace glow {
+
+using ModuleList = std::list<Module *>;
+
+struct PartitionResult {
+  /// List of modules after partitioning. Each module will be assigned to run a
+  /// device by privioner.
+  ModuleList modules_;
+  /// The dependencies of each modules.
+  Map modulesDAG_;
+};
+
+/// Here we assume all the devices are identical.
+struct DeviceInfo {
+  /// Number of the devices.
+  int num;
+  /// Memory constrains of each device. unit GB.
+  int memSize;
+  /// TO DO... communication constrins and so on
+}
+
+/// Partition
+class Partitioner {
+
+  /// Dependency map type.
+  using Map = llvm::DenseMap<Module *, ModuleList>;
+
+  /// The module need to be decomposed.
+  Module module_;
```

Comment:
We probably don't want to copy the Module here, thus a reference or pointer is probably better.